src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  static VALUE_OF_BITES = 50;
+
+  bite(beast) {
+    if (beast instanceof Carnivore || beast.hidden) {
```

Comment:
The logic to check if the beast is a Carnivore or if it's hidden is correct, but it's better to separate these checks for clarity and to potentially provide different feedback for each condition.